students/nDruP/lesson05/mailroom3.py

```diff
@@ -0,0 +1,178 @@
+              ["Rey McScriff", [666.00]],
+              ["Mike Truk", [70935.30, 12546.70, 312.00]],
+              ["Bobson Dugnutt", [1234.56, 789.00]],
+              ["Todd Bonzalez", [715867.83, 10352.07]]]
+for name, money in donor_list:
+    donor_dict[name] = money
+exit_reminder = "Return to the main menu at any time by entering 'exit'"
+
+
+def main_menu():
+    """
+    Prompt user to send a Thank You, Create a report, create letters, or quit.
+    """
+    user_prompt = None
+    valid_prompts = {"1": craft_thank_u,
+                     "2": create_report,
+                     "3": create_letters,
+                     "4": sys.exit}
+    options = list(valid_prompts.keys())
+    while True:
+        print_divider()
+        print("We're a Pyramid Scheme & So Are You! E-Mailroom")
+        print_divider()
+        while user_prompt not in valid_prompts:
+            options_str = ("{}" + (", {}") * (len(options)-1)).format(*options)
+            print(f"Please choose from the following options ({options_str}):")
+            print("1. Send a Thank you")
+            print("2. Create Donor Report")
+            print("3. Send letters to everyone")
+            print("4. Quit")
+            user_prompt = input(">")
+        print_divider()
+        valid_prompts.get(user_prompt)()
+        input("\nPress enter to continue...")
+        user_prompt = None
+
+
+def craft_thank_u():
+    """
+    Prompt for a full name.
+    Prompt->"list": show a list of the donor names
+    Prompt->name not in list: add name to list;
+    Then prompt for donation amount, add amount to donation history
+    Compose and print an email thanking the donor for their donation.
+    Return to main
+    """
+    donor_name = "list"
+    donation_amt = None
+    message = ''
+    while True:
+        print("Let's craft a very personal thank you note for our donor!")
+        print(exit_reminder)
+
+        print("Pull up a list of donor names by entering 'list'")
+        while donor_name == "list":
+            donor_name = input("Enter Donor Name >")
+            if donor_name.lower() == "list":
+                print(("{}\n"*len(donor_dict)).format(*donor_dict))
+        check_exit(donor_name)
+
+        while True:
+            donation_amt = input("Enter their Donation Amount >")
+            check_exit(donation_amt)
+            try:
+                donation_amt = round(float(donation_amt), 2)
+                break
+            except ValueError:
+                print("Invalid input")
+
+        donor_dict[donor_name].append(donation_amt)
+        print_divider()
+        message = (f"Dearest {donor_name},\n"
+                   f"Thank you so much for donation of ${donation_amt:.2f}!\n"
+                   "We will use your donation to create real living Pokemon.\n"
+                   "Sincerely,\n"
+                   f"We're a Pyramid Scheme & so is {donor_name}")
+        print(message)
+        print_divider()
+
+        break
```

Comment:
(unless I'm misunderstanding the flow here) having the `break` here means that while loop will execute once so probably no need for `while True`?